src/services/pcn-router/src/Router.h

```diff
@@ -54,93 +49,82 @@ struct rt_v {
-                    const std::string &nexthop,
+  void replaceRoute(const std::string &network, const std::string &nexthop,
                     const RouteJsonObject &conf) override;
-  void delRoute(const std::string &network, const std::string &netmask,
+  void delRoute(const std::string &network,
                 const std::string &nexthop) override;
   void delRouteList() override;
 
   /// <summary>
-  /// Entry of the ports table
+  /// Entry associated with the ARP table
   /// </summary>
-  std::shared_ptr<Ports> getPorts(const std::string &name) override;
-  std::vector<std::shared_ptr<Ports>> getPortsList() override;
-  void addPorts(const std::string &name, const PortsJsonObject &conf) override;
-  void addPortsList(const std::vector<PortsJsonObject> &conf) override;
-  void replacePorts(const std::string &name,
-                    const PortsJsonObject &conf) override;
-  void delPorts(const std::string &name) override;
-  void delPortsList() override;
+  std::shared_ptr<ArpTable> getArpTable(const std::string &address) override;
+  std::vector<std::shared_ptr<ArpTable>> getArpTableList() override;
+  void addArpTable(const std::string &address,
+                   const ArpTableJsonObject &conf) override;
+  void addArpTableList(const std::vector<ArpTableJsonObject> &conf) override;
+  void replaceArpTable(const std::string &address,
+                       const ArpTableJsonObject &conf) override;
+  void delArpTable(const std::string &address) override;
+  void delArpTableList() override;
 
   // The following methods have been added by hand
 
   // FIXME: the following methods should be protected. Ports should be friend
 
   void add_local_route(const std::string &interface_ip,
-                       const std::string &interface_netmask,
                        const std::string &port_name, const int port_index);
 
   void remove_local_route(const std::string &interface_ip,
-                          const std::string &interface_netmask,
                           const std::string &port_name);
 
-  void delete_route(const std::string &interface_ip,
-                    const std::string &interface_netmask);
+  void delete_route(const std::string &interface_ip);
 
   std::string search_interface_from_nexthop(const std::string &nexthop);
 
   void add_active_nexthop_in_ebpf_map(const std::string &network,
-                                      const std::string &netmask,
                                       const std::string &nexthop,
-                                      int32_t actual_pathcost,
+                                      int32_t new_pathcost,
                                       const int &port_id);
 
-  void remove_route(const std::string &network, const std::string &netmask,
-                    const std::string &nexthop);
+  void remove_route(const std::string &network, const std::string &nexthop);
 
   void remove_all_routes();
 
-  /* Netlink */
+  // Shadow
   std::mutex router_mutex;
```

Comment:
Can we be more specific with respect to "router_mutex"?
I cannot understand what this mutex protect about.